### DISCUSSION TITLE
drop duplicate row from table

### DISCUSSION THREAD

**Discussion by kenn719** [jtablesaw/tablesaw#964] [Q&A]
Hi, do you have plan to include the feature to drop duplicate row from a table based on different column ? Example. Keep the first occurrence or last occurrence similar to python pandas ?

**Comment by lwhite1** [comment]
myTable.dropDuplicateRows() considers all the columns in the table.

**Reply by kenn719** [reply]
Hi 
Thanks for the reply
But did the tablesaw come with the feature to only select certain column as the index for dropping the duplication rows?

**Comment by lwhite1** [comment]
no it doesn't. It would be nice to have, but for now you'd have to write
your own method for that

**Reply by kenn719** [reply]
Do you have the plan to include this feature in future ?

**Comment by lwhite1** [comment]
not currently. but now that there is an open issue for it, i will consider
it in the future.

**Reply by lwhite1** [reply]
Giving it a little more thought, i'm not sure it fits many use-cases. You are basically saying:

> "Give me one random row for each unique combination of these column values Keep the other columns, but I don't care what values are in them."   

 It's basically a (non)random selection. You can do this really easily using Table.splitOn(my_four_column names). Roughly: 

```
    Table c = myTable.emptyCopy();
    for (TableSlice slice : myTable.splitOn("a", "b", "c", "d")) {
        c.append(slice.first(1));
    }
```

Not super efficient that way, but easy enough to write.
